resource-metadata/hypothesis.md

```diff
@@ -0,0 +1,121 @@
+
+For any defined metadata type available for a given resource, all representations of that resource MUST include a Link header pointing to the location of each metadata resource. For example, as defined by the Solid Web Access Control specification, a client can use this mechanism to discover the location of an ACL resource:
+
+Link: <https://example.com/resource?acl>; rel="acl"
+
+### Discovery of Annotated Resource
+
+Certain metadata resource types MAY require the server to include a link back to the annotated resource using an appropriate link relation. For example, LDP defines a bidirectional discovery mechanism for RDF descriptions of NonRDF resources, via Link headers:
+
+Link: <https://example.com/binary?description>; rel="description"
+
+along with:
+
+Link: <https://example.com/binary>; rel="describedby"
+
+For cases where link relations are not defined by IANA, a URL can be used. For example:
+
+Link: <https://example.com/resource?meta>;
+        rel="https://example.org/ns#hasMetadata"
+
+and:
+
+Link: <https://example.com/resource>;
+        rel="https://example.org/ns#isMetadataOf"
+
+String-based link relations, such as in the examples above, must be registered with IANA. But it is also possible to use custom relation types by using a full IRI. The Linked Data Platform, the Linked Data Notification and the Web Annotation specifications make use of full IRIs in the rel attribute.
+
+## Reserved resource metadata types
+
+### Web Access Control
+
+Related Issues:  
+https://github.com/solid/specification/issues/58
+https://github.com/solid/specification/issues/42
+https://github.com/solid/specification/issues/31
+https://github.com/solid/specification/issues/131
+
+ACL resources as defined by [Web Access Control](https://github.com/solid/web-access-control-spec) MUST be supported.
+
+MUST be deleted when resource is deleted.
+
+Authorization Model:
+
+Discovered via ```rel=acl``` and ```rel=http://www.w3.org/ns/solid/terms#resource```
+
+### Resource Description
+
+Resource description provides a general mechanism for resource annotation, such as for providing descriptive metadata for a binary file. MUST be supported. MUST be deleted when resource is deleted.
+
+Authorization Model:
+
+Discovered via ```rel=describedby``` and ```rel=describes```
+
+### Shape Constraint
+
+Enforces conformance of the resource to an associated SHACL or ShEx data shape. MUST be supported. MUST be deleted when resource is deleted.
+
+Authorization Model:
+
+Discovered via ```rel=http://www.w3.org/ns/solid/terms#shape``` and ```rel=http://www.w3.org/ns/solid/terms#resource```
+
+### Server Managed
+
+Server-managed metadata not otherwise included in HTTP headers. MUST be supported.
+
+__Note: This may be too broad to be able to properly determine what the authorization rule, or lifecycle should be. May need to consider breaking this up into specific types.__
```

Comment:
Since we don't expose them over HTTP I don't think authorization rules apply to them. We may just need to define something to make them portable.

Reply:
> I've realized that I don't understand how it currently works, and I still miss some information to understand it.
> 
> The main thing I fail to understand relates to the question: "what is a resource in its own right?"
> 
> I've observed that if you add triples to a container, those triples will be added to `.meta`. In that case, `.meta` is not a resource in its own right, I suppose. You wouldn't interact with it, you'd interact with the container.
> 
> Then, I suppose that if you have an image, and you wish to add metadata to the image, you would do that to a separate resource, and you would get metadata by dereferencing a separate resource. So, in that case, the metadata resource is a resource in its own right. Concievably, you could discriminate between the representations by media type though.
> 
> It would be nice to have clarification around that.

Take a look at latest changes, which stipulate that metadata resources on the same server follow the LDPR interaction model, and then (in implementation patterns) that servers know/maintain which resources are metadata, and can apply special handling to them.

Reply:
Per the discussion in the panel meeting yesterday this was poorly worded and a bit misleading as written because it was incomplete. The next round will complete the section and address this so will resolve for now.